src/expire-tiles.cpp

```diff
@@ -78,41 +78,60 @@ void expire_tiles::from_point_list(geom::point_list_t const &list)
     });
 }
 
-void expire_tiles::from_geometry(geom::geometry_t const &geom)
+void expire_tiles::from_geometry(geom::point_t const &geom)
+{
+    geom::box_t const box = geom::envelope(geom);
+    from_bbox(box);
+}
+
+void expire_tiles::from_geometry(geom::linestring_t const &geom)
+{
+    from_point_list(geom);
+}
+
+void expire_tiles::from_polygon_boundary(geom::polygon_t const &geom)
 {
-    if (geom.srid() != 3857) {
-        return;
+    from_point_list(geom.outer());
+    for (auto const &inner : geom.inners()) {
+        from_point_list(inner);
     }
+}
 
-    if (geom.is_point()) {
-        auto const box = geom::envelope(geom);
-        from_bbox(box);
-    } else if (geom.is_linestring()) {
-        from_point_list(geom.get<geom::linestring_t>());
-    } else if (geom.is_multilinestring()) {
-        for (auto const &list : geom.get<geom::multilinestring_t>()) {
-            from_point_list(list);
-        }
-    } else if (geom.is_polygon() || geom.is_multipolygon()) {
-        auto const box = geom::envelope(geom);
-        if (from_bbox(box)) {
-            if (geom.is_polygon()) {
-                from_point_list(geom.get<geom::polygon_t>().outer());
-                for (auto const &inner : geom.get<geom::polygon_t>().inners()) {
-                    from_point_list(inner);
-                }
-            } else if (geom.is_multipolygon()) {
-                for (auto const &polygon : geom.get<geom::multipolygon_t>()) {
-                    from_point_list(polygon.outer());
-                    for (auto const &inner : polygon.inners()) {
-                        from_point_list(inner);
-                    }
-                }
-            }
+void expire_tiles::from_geometry(geom::polygon_t const &geom)
+{
+    geom::box_t const box = geom::envelope(geom);
+    if (from_bbox(box)) {
+        /* Bounding box too big - just expire tiles on the boundary */
+        from_polygon_boundary(geom);
+    }
+}
+
+void expire_tiles::from_geometry(geom::multipolygon_t const &geom)
+{
+    geom::box_t box;
+    for (auto const &sgeom : geom) {
+        box.extend(sgeom.outer());
+    }
```

Comment:
Isn't that the same as `box = geom::envelope(geom)`?

Reply:
Yes, I missed that one. Changed.